AK/String.cpp

```diff
@@ -256,6 +256,36 @@ ErrorOr<String> String::vformatted(StringView fmtstr, TypeErasedFormatParams& pa
     return builder.to_string();
 }
 
+ErrorOr<Vector<String>> String::split(u32 separator, SplitBehavior split_behavior) const
+{
+    return split_limit(separator, 0, split_behavior);
+}
+
+ErrorOr<Vector<String>> String::split_limit(u32 separator, size_t limit, SplitBehavior split_behavior) const
+{
+    Vector<String> result;
+
+    if (is_empty())
+        return result;
+
+    bool keep_empty = has_flag(split_behavior, SplitBehavior::KeepEmpty);
+
+    size_t substring_start = 0;
+    for (size_t i = 0; i < bytes().size() && (result.size() + 1) != limit; ++i) {
+        u32 current_code_point = bytes_as_string_view()[i];
+        if (current_code_point == separator) {
```

Comment:
So here, you're now mixing code points and bytes, which are not the same. A code point can be encoded as multiple bytes. 

For example, if I were to try `String("ωΩω"sv).split(0x03A9u)`, this would never find the capital Ω (U+03A9), as `current_code_point` will be just the first of its 2 UTF-8 encoded bytes (0xCE 0xA9).

 Therefore we should be iterating over `String::code_points()` rather than `String::bytes()` or `String::bytes_as_string_view()`. I recommend beefing up the unit tests to catch errors like this, i.e. splitting on non-ASCII code points.

Reply:
Thanks for the guidance. Just updated the PR.